rest/route/github.go

```diff
@@ -306,6 +319,135 @@ func (gh *githubHookApi) AddIntentForPR(pr *github.PullRequest) error {
+	}
+	catcher := grip.NewBasicCatcher()
+	for _, pRef := range projectRefs {
+		// if a version for this revision exists for this project, add tag
+		existingVersion, err := gh.sc.FindVersionByProjectAndRevision(pRef.Identifier, hash)
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem finding version for project '%s' with revision '%s' to add tag '%s'",
+				pRef.Identifier, hash, tag.Tag))
+			continue
+		}
+		if existingVersion == nil {
+			grip.Info(message.Fields{
+				"source":  "github hook",
+				"message": "no version to add tag to",
+				"branch":  pRef.Branch,
+				"repo":    pRef.Repo,
+				"owner":   pRef.Owner,
+				"tag":     tag,
+			})
+			continue
+		}
+
+		if err = gh.sc.AddGitTagToVersion(existingVersion.Id, tag); err != nil {
+			catcher.Add(errors.Wrapf(err, "problem adding tag '%s' to version '%s''", tag.Tag, existingVersion.Id))
+			continue
+		}
+
+		// TODO: only do this if project doesn't have a separate yaml defined
+		hasAliases, err := gh.sc.HasMatchingGitTagAlias(pRef.Identifier, tag.Tag)
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem finding aliases for project '%s'", pRef.Identifier))
+			continue
+		}
+		if !hasAliases {
+			continue
+		}
+
+		p, pp, err := gh.sc.LoadProjectForVersion(existingVersion, pRef.Identifier)
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem getting project for  '%s'", pRef.Identifier))
+			continue
+		}
+		info := &model.ProjectInfo{
+			Ref:                 &pRef,
+			Project:             p,
+			IntermediateProject: pp,
+		}
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem populating project info for tag '%s' for project '%s'", tag.Tag, pRef.Identifier))
+			continue
+
+		}
+		metadata := model.VersionMetadata{
+			Revision: model.Revision{
+				Author:          event.GetSender().GetLogin(),
+				AuthorGithubUID: int(event.GetSender().GetID()),
+				AuthorEmail:     event.GetSender().GetEmail(),
+				Revision:        hash,
+			},
+			IsAdHoc: true,
+			Alias:   evergreen.GitTagAlias,
+			GitTag:  tag,
+		}
+		_, err = gh.sc.CreateVersionFromConfig(ctx, info, metadata, true)
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem creating version for tag '%s' for project '%s'", tag.Tag, pRef.Identifier))
+			continue
+		}
+	}
+	grip.Error(message.WrapError(catcher.Resolve(), message.Fields{
+		"source":  "github hook",
+		"msg_id":  gh.msgID,
+		"event":   gh.eventType,
+		"owner":   ownerAndRepo[0],
+		"repo":    ownerAndRepo[1],
+		"tag":     tag,
+		"message": "errors updating/creating versions for git tag",
+	}))
+	return nil
+}
+
+func validatePushTagEvent(event *github.PushEvent) error {
+	if len(strings.Split(event.Repo.GetFullName(), "/")) != 2 {
+		return errors.New("Repo name is invalid (expected [owner]/[repo])")
+	}
+
+	// check if tag is valid for project
+	if event.GetRef() == "" {
+		return errors.New("Base ref is empty")
+	}
+
+	if event.GetSender().GetLogin() == "" || event.GetSender().GetID() == 0 {
```

Comment:
this is duplicated

Reply:
Do you mean duplicated elsewhere or duplicated below? (One is for pusher, and one is for sender)

Reply:
oops can't read